apps/web/vibes/soul/sections/cart/index.tsx

```diff
@@ -53,71 +46,62 @@ interface CartState<LineItem extends CartLineItem> {
   lastResult: SubmissionResult | null;
 }
 
+export interface Cart<LineItem extends CartLineItem> {
+  lineItems: LineItem[];
+  summaryLineItems: CartSummaryLineItem[];
+  total: string;
+}
+
 interface CartProps<LineItem extends CartLineItem> {
   title?: string;
-  lineItems: LineItem[] | Promise<LineItem[]>;
-  summary: CartSummary;
-  emptyState: CartEmptyState;
+  summaryTitle?: string;
+  emptyState?: CartEmptyState;
   lineItemAction: Action<CartState<LineItem>, FormData>;
   checkoutAction: Action<SubmissionResult | null, FormData>;
+  checkoutLabel?: string;
```

Comment:
I think we're going to need to split this component into two files because we want the labels to be streamable as well since those are localized. Then, we'll want the component that uses `Stream` to be a RSC so we can `Promise.all` all streamable values

Reply:
Sounds good. Why would we stream in a labels instead of just awaiting them btw?

Reply:
It's what we've been doing so far for localized labels. Perhaps it's unnecessary and this will all probably change once we implement label i18n